Assets/Scripts/Audio/AudioController.cs

```diff
@@ -169,6 +169,19 @@ public void SetMusicVolume(int volume)
 		onMusicVolumeChange(this.musicVolume);
 	}
 
+    //BP Does the sound already have a channel (ie an audiosource on 'Audio' gameObject)?
+    bool ClipHasChannel(AudioFile file)
+    {
+        foreach(AudioSource channel in channels.Values)
```

Comment:
This took a bit to figure out but I ended up creating a new `HashSet<AudioClip> activeSFX` that seems to work great. 👍